source/opt/basic_block.cpp

```diff
@@ -101,13 +109,31 @@ void BasicBlock::ForMergeAndContinueLabel(
   --ii;
   if (ii == insts_.begin()) return;
   --ii;
-  if (ii->opcode() == SpvOpSelectionMerge || 
-      ii->opcode() == SpvOpLoopMerge)
-    ii->ForEachInId([&f](const uint32_t* idp) {
-      f(*idp);
-    });
+  if (ii->opcode() == SpvOpSelectionMerge || ii->opcode() == SpvOpLoopMerge)
+    ii->ForEachInId([&f](const uint32_t* idp) { f(*idp); });
+}
+
+uint32_t BasicBlock::MergeBlockIdIfAny(uint32_t* cbid) {
```

Comment:
Not that this needs to be done in this PR, but it is worth splitting this into two function?  One to get the merge block, and another to get the continue block?  It does not look like there is any looping going on.  I won't fight for it if others disagree.

Reply:
Yeah, this function bugged me a bit because it does two distinct things.  I've added BasicBlock::ContinueBlockIdIfAny().